BUG: more numpy repr issues

I'm not sure why these surfaced up yet again (sometimes in the past ~5days), but it's not blocking the release I already tagged locally 😭 

cc @pllim and will tag Sebastian as well once it's clear it's due to recent numpy changes rather than astropy.

```
platform darwin -- Python 3.11.0, pytest-7.4.0, pluggy-1.2.0

Running tests with pyvo version 1.5.dev175+g6656130.
Running tests in pyvo docs.

Date: 2023-08-16T13:34:18

Platform: macOS-12.5.1-arm64-arm-64bit

Executable: /Users/bsipocz/.pyenv/versions/3.11.0/bin/python3.11

Full Python Version: 
3.11.0 (main, Nov 27 2022, 14:22:00) [Clang 14.0.0 (clang-1400.0.29.202)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 2.0.0.dev0+git20230815.4387e03
Astropy: 5.3.2
requests: 2.28.1

Using Astropy options: remote_data: none.

rootdir: /Users/bsipocz/munka/devel/pyvo
configfile: setup.cfg
testpaths: pyvo, docs
plugins: doctestplus-1.0.0, remotedata-0.4.0, rerunfailures-10.3, nbval-0.10.0, hypothesis-6.58.1, mock-3.10.0, dependency-0.5.1, filter-subpackage-0.1.2, requests-mock-1.10.0, astropy-header-0.2.2, astropy-0.10.0, cov-4.0.0, openfiles-0.5.0, anyio-3.6.2, arraydiff-0.5.0
collected 326 items                                                                                                                      

pyvo/auth/tests/test_auth.py ......                                                                                                [  1%]
pyvo/dal/mimetype.py .                                                                                                             [  2%]
pyvo/dal/tests/test_adhoc.py .....                                                                                                 [  3%]
pyvo/dal/tests/test_datalink.py ........                                                                                           [  6%]
pyvo/dal/tests/test_mimetype.py .                                                                                                  [  6%]
pyvo/dal/tests/test_monkeypatch.py .                                                                                               [  6%]
pyvo/dal/tests/test_params.py .F.....                                                                                              [  8%]
pyvo/dal/tests/test_query.py .................................F.....                                                               [ 20%]
pyvo/dal/tests/test_scs.py ...                                                                                                     [ 21%]
pyvo/dal/tests/test_sia.py ...                                                                                                     [ 22%]
pyvo/dal/tests/test_sia2.py ..........                                                                                             [ 25%]
pyvo/dal/tests/test_sia2_remote.py ssssssssssssssssss                                                                              [ 31%]
pyvo/dal/tests/test_sla.py ..                                                                                                      [ 31%]
pyvo/dal/tests/test_ssa.py ..                                                                                                      [ 32%]
pyvo/dal/tests/test_tap.py ................s....                                                                                   [ 38%]
pyvo/io/uws/tests/test_job.py ..                                                                                                   [ 39%]
pyvo/io/vosi/tests/test_availability.py .                                                                                          [ 39%]
pyvo/io/vosi/tests/test_capabilities.py .....................                                                                      [ 46%]
pyvo/io/vosi/tests/test_tables.py .......................x..........                                                               [ 56%]
pyvo/registry/rtcons.py ...                                                                                                        [ 57%]
pyvo/registry/tests/test_regtap.py .............................................sssss.....sssss                                    [ 76%]
pyvo/registry/tests/test_rtcons.py .....F..........................FF.FFFFFF.FF........                                            [ 92%]
pyvo/utils/tests/test_http.py .                                                                                                    [ 92%]
pyvo/utils/tests/test_prototype.py ........                                                                                        [ 94%]
pyvo/utils/tests/test_url.py .                                                                                                     [ 95%]
pyvo/utils/tests/test_vocabularies_remote.py ssssss                                                                                [ 96%]
pyvo/utils/xml/tests/test_elements.py .......                                                                                      [ 99%]
docs/index.rst s                                                                                                                   [ 99%]
docs/dal/index.rst .                                                                                                               [ 99%]
docs/registry/index.rst .                                                                                                          [100%]

================================================================ FAILURES ================================================================
_____________________________________________________________ test_serialize _____________________________________________________________

    @pytest.mark.usefixtures('proc')
    @pytest.mark.filterwarnings("ignore::astropy.io.votable.exceptions.W06")
    @pytest.mark.filterwarnings("ignore::astropy.io.votable.exceptions.W48")
    @pytest.mark.filterwarnings("ignore::astropy.io.votable.exceptions.E02")
    def test_serialize():
        datalink = DatalinkResults.from_result_url('http://example.com/proc')
        proc_dl = datalink[0]
        input_params = {param.name: param for param in proc_dl.input_params}
    
        polygon_conv = get_converter(
            find_param_by_keyword('polygon', input_params))
        circle_conv = get_converter(
            find_param_by_keyword('circle', input_params))
        scale_conv = get_converter(
            find_param_by_keyword('scale', input_params))
        kind_conv = get_converter(
            find_param_by_keyword('kind', input_params))
    
>       assert polygon_conv.serialize((1, 2, 3)) == "1 2 3"
E       AssertionError: assert 'np.float64(1....float64(3.0)' == '1 2 3'
E         - 1 2 3
E         + np.float64(1.0) np.float64(2.0) np.float64(3.0)

pyvo/dal/tests/test_params.py:139: AssertionError
__________________________________________________________ TestRecord.test_repr __________________________________________________________

self = <pyvo.dal.tests.test_query.TestRecord object at 0x15d551390>

    def test_repr(self):
        record = DALResults.from_result_url(
            'http://example.com/query/basic')[0]
        truth = 'Illuminatus'
>       assert repr(record) == repr((23, truth))
E       assert "(np.int64(23...lluminatus'))" == "(23, 'Illuminatus')"
E         - (23, 'Illuminatus')
E         + (np.int64(23), np.str_('Illuminatus'))

pyvo/dal/tests/test_query.py:437: AssertionError
_______________________________________________________ TestSQLLiterals.test_numpy _______________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSQLLiterals object at 0x15d544990>
literals = {'aFloat': '50000000.0', 'aString': "'some harmless stuff'", 'anInt': '210', 'bytes': "'keep this ascii for now'", ...}

    def test_numpy(self, literals):
>       assert float(literals["numpyStuff"]) - 23.7 < 1e-10
E       ValueError: could not convert string to float: 'np.float64(23.7)'

pyvo/registry/tests/test_rtcons.py:57: ValueError
__________________________________________________ TestSpatialConstraint.test_SkyCoord ___________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpatialConstraint object at 0x15d65a3d0>

    def test_SkyCoord(self):
        cons = registry.Spatial(SkyCoord(3 * u.deg, -30 * u.deg))
>       assert cons.get_search_condition() == "1 = CONTAINS(MOC(6, POINT(3.0, -30.0)), coverage)"
E       AssertionError: assert '1 = CONTAINS...)), coverage)' == '1 = CONTAINS...)), coverage)'
E         - 1 = CONTAINS(MOC(6, POINT(3.0, -30.0)), coverage)
E         + 1 = CONTAINS(MOC(6, POINT(np.float64(3.0), np.float64(-30.0))), coverage)
E         ?                           +++++++++++   +  +++++++++++       +

pyvo/registry/tests/test_rtcons.py:221: AssertionError
_______________________________________________ TestSpatialConstraint.test_SkyCoord_Circle _______________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpatialConstraint object at 0x15d659d10>

    def test_SkyCoord_Circle(self):
        cons = registry.Spatial((SkyCoord(3 * u.deg, -30 * u.deg), 3))
>       assert cons.get_search_condition() == "1 = CONTAINS(MOC(6, CIRCLE(3.0, -30.0, 3)), coverage)"
E       AssertionError: assert '1 = CONTAINS...)), coverage)' == '1 = CONTAINS...)), coverage)'
E         - 1 = CONTAINS(MOC(6, CIRCLE(3.0, -30.0, 3)), coverage)
E         + 1 = CONTAINS(MOC(6, CIRCLE(np.float64(3.0), np.float64(-30.0), 3)), coverage)
E         ?                            +++++++++++   +  +++++++++++     +

pyvo/registry/tests/test_rtcons.py:226: AssertionError
_________________________________________________ TestSpectralConstraint.test_energy_eV __________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpectralConstraint object at 0x15d65b4d0>

    def test_energy_eV(self):
        cons = registry.Spectral(5 * u.eV)
>       assert cons.get_search_condition() == "8.01088317e-19 BETWEEN spectral_start AND spectral_end"
E       AssertionError: assert 'np.float64(8... spectral_end' == '8.01088317e-... spectral_end'
E         - 8.01088317e-19 BETWEEN spectral_start AND spectral_end
E         + np.float64(8.01088317e-19) BETWEEN spectral_start AND spectral_end
E         ? +++++++++++              +

pyvo/registry/tests/test_rtcons.py:240: AssertionError
______________________________________________ TestSpectralConstraint.test_energy_interval _______________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpectralConstraint object at 0x15c47f4d0>

    def test_energy_interval(self):
        cons = registry.Spectral((1e-10 * u.erg, 2e-10 * u.erg))
>       assert cons.get_search_condition() == (
            "1 = ivo_interval_overlaps(spectral_start, spectral_end, 1e-17, 2e-17)")
E       AssertionError: assert '1 = ivo_inte...oat64(2e-17))' == '1 = ivo_inte...1e-17, 2e-17)'
E         Skipping 46 identical leading characters in diff, use -v to show
E         - tral_end, 1e-17, 2e-17)
E         + tral_end, np.float64(1e-17), np.float64(2e-17))

pyvo/registry/tests/test_rtcons.py:244: AssertionError
_________________________________________________ TestSpectralConstraint.test_wavelength _________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpectralConstraint object at 0x15d65ba10>

    def test_wavelength(self):
        cons = registry.Spectral(5000 * u.Angstrom)
>       assert cons.get_search_condition() == "3.9728917142978567e-19 BETWEEN spectral_start AND spectral_end"
E       AssertionError: assert 'np.float64(3... spectral_end' == '3.9728917142... spectral_end'
E         - 3.9728917142978567e-19 BETWEEN spectral_start AND spectral_end
E         + np.float64(3.9728917142978567e-19) BETWEEN spectral_start AND spectral_end
E         ? +++++++++++                      +

pyvo/registry/tests/test_rtcons.py:249: AssertionError
____________________________________________ TestSpectralConstraint.test_wavelength_interval _____________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpectralConstraint object at 0x15d65af10>

    def test_wavelength_interval(self):
        cons = registry.Spectral((20 * u.cm, 22 * u.cm))
>       assert (cons.get_search_condition()
                == "1 = ivo_interval_overlaps(spectral_start, spectral_end,"
                " 9.932229285744642e-25, 9.029299350676949e-25)")
E       AssertionError: assert '1 = ivo_inte...0676949e-25))' == '1 = ivo_inte...50676949e-25)'
E         Skipping 46 identical leading characters in diff, use -v to show
E         - tral_end, 9.932229285744642e-25, 9.029299350676949e-25)
E         + tral_end, np.float64(9.932229285744642e-25), np.float64(9.029299350676949e-25))
E         ?           +++++++++++                     +  +++++++++++                      +

pyvo/registry/tests/test_rtcons.py:253: AssertionError
_________________________________________________ TestSpectralConstraint.test_frequency __________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpectralConstraint object at 0x15d65aa50>

    def test_frequency(self):
        cons = registry.Spectral(2 * u.GHz)
>       assert (cons.get_search_condition()
                == "1.32521403e-24 BETWEEN spectral_start AND spectral_end")
E       AssertionError: assert 'np.float64(1... spectral_end' == '1.32521403e-... spectral_end'
E         - 1.32521403e-24 BETWEEN spectral_start AND spectral_end
E         + np.float64(1.32521403e-24) BETWEEN spectral_start AND spectral_end
E         ? +++++++++++              +

pyvo/registry/tests/test_rtcons.py:259: AssertionError
_____________________________________________ TestSpectralConstraint.test_frequency_interval _____________________________________________

self = <pyvo.registry.tests.test_rtcons.TestSpectralConstraint object at 0x15d65a410>

    def test_frequency_interval(self):
        cons = registry.Spectral((88 * u.MHz, 102 * u.MHz))
>       assert (cons.get_search_condition()
                == "1 = ivo_interval_overlaps(spectral_start, spectral_end,"
                " 5.830941732e-26, 6.758591553e-26)")
E       AssertionError: assert '1 = ivo_inte...8591553e-26))' == '1 = ivo_inte...58591553e-26)'
E         Skipping 46 identical leading characters in diff, use -v to show
E         - tral_end, 5.830941732e-26, 6.758591553e-26)
E         + tral_end, np.float64(5.830941732e-26), np.float64(6.758591553e-26))
E         ?           +++++++++++               +  +++++++++++                +

pyvo/registry/tests/test_rtcons.py:264: AssertionError
________________________________________________ TestTemporalConstraint.test_single_time _________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestTemporalConstraint object at 0x15d658150>

    def test_single_time(self):
        cons = registry.Temporal(Time('2022-01-10'))
>       assert (cons.get_search_condition()
                == "59589.0 BETWEEN time_start AND time_end")
E       AssertionError: assert 'np.float64(5... AND time_end' == '59589.0 BETW... AND time_end'
E         - 59589.0 BETWEEN time_start AND time_end
E         + np.float64(59589.0) BETWEEN time_start AND time_end
E         ? +++++++++++       +

pyvo/registry/tests/test_rtcons.py:277: AssertionError
_______________________________________________ TestTemporalConstraint.test_time_interval ________________________________________________

self = <pyvo.registry.tests.test_rtcons.TestTemporalConstraint object at 0x15c5c2850>

    def test_time_interval(self):
        cons = registry.Temporal((Time(2459000, format='jd'),
                                  Time(59002, format='mjd')))
>       assert (cons.get_search_condition()
                == "1 = ivo_interval_overlaps(time_start, time_end, 58999.5, 59002.0)")
E       AssertionError: assert '1 = ivo_inte...t64(59002.0))' == '1 = ivo_inte...9.5, 59002.0)'
E         Skipping 38 identical leading characters in diff, use -v to show
E         - time_end, 58999.5, 59002.0)
E         + time_end, np.float64(58999.5), np.float64(59002.0))

pyvo/registry/tests/test_rtcons.py:283: AssertionError
======================================================== short test summary info =========================================================
SKIPPED [35] ../../../.pyenv/versions/3.11.0/lib/python3.11/site-packages/pytest_remotedata/plugin.py:85: need --remote-data option to run
SKIPPED [1] ../../../.pyenv/versions/3.11.0/lib/python3.11/site-packages/_pytest/doctest.py:455: all tests skipped by +SKIP option
XFAIL pyvo/io/vosi/tests/test_tables.py::TestTables::test_wrong_flag
FAILED pyvo/dal/tests/test_params.py::test_serialize - AssertionError: assert 'np.float64(1....float64(3.0)' == '1 2 3'
FAILED pyvo/dal/tests/test_query.py::TestRecord::test_repr - assert "(np.int64(23...lluminatus'))" == "(23, 'Illuminatus')"
FAILED pyvo/registry/tests/test_rtcons.py::TestSQLLiterals::test_numpy - ValueError: could not convert string to float: 'np.float64(23.7)'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpatialConstraint::test_SkyCoord - AssertionError: assert '1 = CONTAINS...)), coverage)' == '1 = CONTAINS...)), coverage)'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpatialConstraint::test_SkyCoord_Circle - AssertionError: assert '1 = CONTAINS...)), coverage)' == '1 = CONTAINS...)), coverage)'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpectralConstraint::test_energy_eV - AssertionError: assert 'np.float64(8... spectral_end' == '8.01088317e-... spectral_end'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpectralConstraint::test_energy_interval - AssertionError: assert '1 = ivo_inte...oat64(2e-17))' == '1 = ivo_inte...1e-17, 2e-17)'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpectralConstraint::test_wavelength - AssertionError: assert 'np.float64(3... spectral_end' == '3.9728917142... spectral_end'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpectralConstraint::test_wavelength_interval - AssertionError: assert '1 = ivo_inte...0676949e-25))' == '1 = ivo_inte...50676949e-25)'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpectralConstraint::test_frequency - AssertionError: assert 'np.float64(1... spectral_end' == '1.32521403e-... spectral_end'
FAILED pyvo/registry/tests/test_rtcons.py::TestSpectralConstraint::test_frequency_interval - AssertionError: assert '1 = ivo_inte...8591553e-26))' == '1 = ivo_inte...58591553e-26)'
FAILED pyvo/registry/tests/test_rtcons.py::TestTemporalConstraint::test_single_time - AssertionError: assert 'np.float64(5... AND time_end' == '59589.0 BETW... AND time_end'
FAILED pyvo/registry/tests/test_rtcons.py::TestTemporalConstraint::test_time_interval - AssertionError: assert '1 = ivo_inte...t64(59002.0))' == '1 = ivo_inte...9.5, 59002.0)'
========================================= 13 failed, 276 passed, 36 skipped, 1 xfailed in 1.86s ==========================================
```
```

--- a/pyvo/conftest.py
+++ b/pyvo/conftest.py
@@ -17,7 +17,7 @@ except ImportError:
     ASTROPY_HEADER = False
 
 # Keep this until we require numpy to be >=2.0
-if minversion(np, "2.0.0.dev0+151"):
+if minversion(np, "2.0.0.dev0+git20230726"):
     np.set_printoptions(legacy="1.25")
 
 

